### PR TITLE
[255] reduced number of dates shown on x axis to allow for even spacing

### PR DESCRIPTION
**Pull request by JohnAGill**
This reduces the number of dates shown from 6 to 5 so that the spacing between them is always consistent

issue: [#255 ](https://github.com/project-vagabond/covid-green-app/issues/255)

### CODE REVIEW

**Review by ShaunBaker — CHANGES_REQUESTED** [review]
I dont think this is a valid fix.

Desired data is to show the past week and just dropping a day is not sufficient.

**Comment by AlanSl** [comment]
@ShaunBaker in NY now it's 30 days of data not one week. Currently shows 7 day intervals

Also, there's a remaining bug on the server side that is giving us 29 days of data instead of 30 which is part of the reason the last date looks so off.

**Comment by AlanSl** [comment]
I like how the current fix applied by @JohnAGill looks currently - but unfortunately it's going to break as soon as we actually get 30 days of data. There will be two labels clashing at the end like this:



30 days of data: `6` looks best. 29 days of data: `7` looks best. And we have no idea when the 30 day fix is finally going to come through. 

It's not a great situation.

@JohnAGill can you maybe make it conditional, for example take it out of props and set the value to `6` if `axisData.length` is 29 and otherwise `6` as it was before. Also add a comment explaining what's going on because it'll look odd that something like that is hardcoded but I think it's the best we can do.

The alternative is import `d3.ticks` and let it figure it out, but it's quirky.

File: src/components/molecules/bar-chart.tsx [hunk 1]
```diff
@@ -70,13 +70,12 @@ export const TrackerBarChart: FC<TrackerBarChartProps> = ({
...
   yMin = 5,
   ySuffix = '',
-  intervalsCount = 6,
   primaryColor = '#CD4000',
   secondaryColor = '#ACAFC4',
   backgroundColor = colors.white
 }) => {
   const {t} = useTranslation();
-
+  const intervalsCount = axisData.length > 30 ? 7 : 6;
```
Comment (AlanSl) [inline src/components/molecules/bar-chart.tsx]:
```suggestion
  const intervalsCount = axisData.length < 30 ? 7 : 6;
```
Issue is with 29 and 30, not 30 and 31

**Review by AlanSl — APPROVED** [review]

**Comment by AlanSl** [comment]
I've fixed the direction of the `<` and the merge conflict so we can get it in.

Antoine's got a new design for the x axis here anyway so this'll just be a temporary thing to stop the x-axis looking broken while we're still stuck with just 29 days of data.

**Review by floridemai — APPROVED** [review]

**Pull request merged**